### ISSUE TITLE
Contex.parseInlineString ignore extra code

### ISSUE THREAD

**Issue by ncannasse** [HaxeFoundation/haxe#11368]
When doing:
```haxe
haxe.macro.Context.parseInlineSring("p:true,v:0", haxe.macro.Context.currentPos())
```

we get a `CIdent("p")` result instead of an error, without a way to be able to tell if the whole code has been parsed.

**Comment by kLabz** [comment]
Same in string interpolation btw

**Comment by Simn** [comment]
This is likely a regression from #9487. String parsing now uses the actual parser functions as entry points, and as long as those can do their thing nobody complains.

I think the only reason this fails properly for normal parsing is because `parse_type_decls` checks for the `Eof` token and fails if there's something else. We can do something similar in `parse_string`, but we might have to be careful because `Eof` might have already been consumed. Not sure if it keeps appearing at the end of a token stream.

**Comment by Simn** [comment]
This fails with `Unexpected :` now. The position looks strange to me but that's probably just from the `currentPos` usage. Still need a test.

The string interpolation version is #10287.

**Comment by Simn** [comment]
This change actually fixed #10287 as well because it goes through the same mechanism.
